driver/ppm_events_public.h

```diff
@@ -1421,6 +1421,18 @@ struct ppm_evt_hdr {
 	uint16_t type; /* the event type */
 	uint32_t nparams; /* the number of parameters of the event */
 };
+
+/// Event header as stored in scap files. if PPM_ENABLE_SENTINEL is not defined,
+/// it must be identical to ppm_evt_hdr. Otherwise, it must match the *end*
+/// of ppm_evt_hdr (scap files don't have the sentinel at the beginning of the struct)
+/// see scap_next_offline() in scap_savefile.c
+struct ppm_scap_evt_hdr {
```

Comment:
I worry about diversion between this struct and ppm_evt_hdr. I suggest adding some compile-time checks to make sure everything's done properly:

>#ifdef PPM_ENABLE_SENTINEL
>#define PPM_STRUCT_DIFF sizeof(uint32_t)
>#else
>#define PPM_STRUCT_DIFF 0
>#endif
>#if sizeof(ppm_scap_evt_hdr) - sizeof(ppm_evt_hdr) != PPM_STRUCT_DIFF
>#error Mismatch between ppm_scap_evt_hdr and ppm_evt_hdr
>#endif


I think something like that will work

>

Reply:
I have the same concern as @nathan-b . Another way to avoid this would be to directly populate ppm_evt_hdr with ppm_scap_evt_hdr as a member and get rid of lines 1418-1422. But that will require changes to all files where ppm_evt_hdr is used.